telemetry/peer_count.go

```diff
@@ -2,66 +2,84 @@ package telemetry
 
 import (
 	"database/sql"
+	"encoding/json"
+	"fmt"
 	"time"
+
+	"github.com/status-im/dev-telemetry/pkg/types"
 )
 
 type PeerCount struct {
-	ID            int    `json:"id"`
-	CreatedAt     int64  `json:"createdAt"`
-	Timestamp     int64  `json:"timestamp"`
-	NodeName      string `json:"nodeName"`
-	NodeKeyUid    string `json:"nodeKeyUid"`
-	PeerID        string `json:"peerId"`
-	PeerCount     int    `json:"peerCount"`
-	StatusVersion string `json:"statusVersion"`
+	data types.PeerCount
 }
 
-func (r *PeerCount) put(db *sql.DB) error {
+func (r *PeerCount) process(db *sql.DB, errs *MetricErrors, data *types.TelemetryRequest) error {
+	if err := json.Unmarshal(*data.TelemetryData, &r.data); err != nil {
+		errs.Append(data.Id, fmt.Sprintf("Error decoding peer count: %v", err))
+		return err
+	}
+
 	stmt, err := db.Prepare("INSERT INTO peerCount (timestamp, nodeName, nodeKeyUid, peerId, peerCount, statusVersion, createdAt) VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING id;")
 	if err != nil {
 		return err
 	}
 
 	defer stmt.Close()
 
-	r.CreatedAt = time.Now().Unix()
+	r.data.CreatedAt = time.Now().Unix()
 	lastInsertId := 0
-	err = stmt.QueryRow(r.Timestamp, r.NodeName, r.NodeKeyUid, r.PeerID, r.PeerCount, r.StatusVersion, r.CreatedAt).Scan(&lastInsertId)
+	err = stmt.QueryRow(
+		r.data.Timestamp,
+		r.data.NodeName,
+		r.data.NodeKeyUid,
+		r.data.PeerID,
+		r.data.PeerCount,
+		r.data.StatusVersion,
+		r.data.CreatedAt,
+	).Scan(&lastInsertId)
 	if err != nil {
+		errs.Append(data.Id, fmt.Sprintf("Error saving peer count: %v", err))
```

Comment:
I wonder if we should return the details of the error to the user, or if instead we should log the `err`, and just return `error saving peer count`.

Reply:
I guess this depends on where is it better for us to see the errors: clients running status-go, the telemetry server, or both? I think we're most likely to look at the status-go logs from running status-desktop

Reply:
Yeah, it is a tough question - returning the error leaks implementation details to clients - a) they usually don't need to know as they cannot do anything about it, b) it might be dangerous from the security perspective

That said, we'd need to make sure we have good access to Telemetry server logs otherwise to be able to spot the errors and reasons there.

Also, I would prefer not change it in this PR - how about a new issue "Review error handling and reporting in telemetry server and client"?

Reply:
> That said, we'd need to make sure we have good access to Telemetry server logs otherwise to be able to spot the errors and reasons there.

This is possible to do via kibana: https://kibana.infra.status.im/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_a=(columns:!(),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:ffcc22b0-0116-11ed-9719-cdd3b483481c,key:logsource,negate:!f,params:(query:node-01.he-eu-hel1.tmetry.misc),type:phrase),query:(match_phrase:(logsource:node-01.he-eu-hel1.tmetry.misc))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:ffcc22b0-0116-11ed-9719-cdd3b483481c,key:program,negate:!f,params:(query:docker%2Ftelemetry-srv),type:phrase),query:(match_phrase:(program:docker%2Ftelemetry-srv)))),index:ffcc22b0-0116-11ed-9719-cdd3b483481c,interval:auto,query:(language:kuery,query:''),sort:!(!('@timestamp',desc)))